os-server/src/main/java/app/openschool/course/api/mapper/CourseMapper.java

```diff
@@ -24,4 +35,58 @@ public static CourseDto toCourseDto(Course course) {
     return new CourseDto(
         course.getTitle(), course.getRating(), course.getDifficulty().getTitle(), keywords);
   }
+
+  public static CourseInfoDto toCourseInfoDto(Course course) {
+    return new CourseInfoDto(
+        course.getTitle(),
+        course.getDescription(),
+        course.getGoal(),
+        getCourseInfoModuleDtoSet(course),
+        getCourseInfoMentorDto(course),
+        course.getRating(),
+        course.getEnrolledCourses().size(),
+        course.getDifficulty().getTitle(),
+        course.getLanguage().getTitle(),
+        getCourseDuration(course));
+  }
+
+  private static long getCourseDuration(Course course) {
+    return course.getModules().stream()
+        .flatMapToLong(
+            module -> module.getModuleItems().stream().mapToLong(ModuleItem::getEstimatedTime))
+        .sum();
+  }
+
+  private static CourseInfoMentorDto getCourseInfoMentorDto(Course course) {
+    return new CourseInfoMentorDto(
+        course.getMentor().getName(),
+        course.getMentor().getSurname(),
+        course.getMentor().getUserImgPath(),
+        course.getMentor().getLinkedinPath());
+  }
+
+  private static Set<CourseInfoModuleDto> getCourseInfoModuleDtoSet(Course course) {
+    return course.getModules().stream()
+        .map(
+            (module ->
+                new CourseInfoModuleDto(module.getTitle(), getCourseInfoModuleItemDtoSet(module))))
+        .collect(Collectors.toSet());
+  }
+
+  private static Set<CourseInfoModuleItemDto> getCourseInfoModuleItemDtoSet(Module module) {
+    return module.getModuleItems().stream()
+        .map(
+            moduleItem ->
+                new CourseInfoModuleItemDto(moduleItem.getModuleItemType(), moduleItem.getLink()))
+        .collect(Collectors.toSet());
+  }
+
+  public static EnrolledCourse toEnrolledCourse(Course course, User user) {
```

Comment:
please arrange code by the following principle public methods first then private methods

Reply:
however this is not mapper logic, this should be in the service